chia/wallet/puzzles/dao_proposal_timer.clsp

```diff
@@ -40,6 +40,11 @@
   ; proposal_yes_votes
   ; proposal_total_votes
   ; proposal_innerpuzhash
+  ; [REVIEW - optimization] We're passing in a _lot_ of stuff here to simply validate the proposal timelock is correct
+  ; [^] Could we either a) curry in the timelock to this puzzle
+  ; [^] b) have the proposal announce the timelock to us and verify it that way?
+  ; [REVIEW - optimization] Similar vein as above, but couldn't the proposal also verify our parent ID since it already has access to it's own singleton info?
```

Comment:
Our parent is the proposal when it had 0 votes. We assert this so we know that the timer was not cheated by being made earlier. We don't know what the timelock amount is at this time though.

Reply:
Right but the proposal also knows what it's ID was when it had zero votes so it could do ASSERT_COIN_ANNOUNCEMENT and check our parent for us and then we don't need to curry in any of the inner puzzle stuff, this puzzle becomes "I self destruct w/ a relative timelock when a curried singleton tells me to".